spec/lib/reports/products_and_inventory_report_spec.rb

```diff
@@ -273,6 +267,10 @@ module ProductsAndInventory
         let(:report) do
           AllProducts.new user, { fields_to_hide: [] }
         end
+        # Creating a variant directly would create a product with another variant with no supplier
+        # which breaks the test
```

Comment:
So yes any call to create(:variant) actually generate two variants, I could not find a way to prevent that. But I added a comment in the variant factory to explain the issue see : https://github.com/openfoodfoundation/openfoodnetwork/pull/12562/commits/afae083b1d8e00671086f66a60a9149aee75d566